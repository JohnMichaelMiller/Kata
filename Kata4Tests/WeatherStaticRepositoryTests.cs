using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class WeatherStaticRepositoryTests
    {
        [TestMethod()]
        public void WeatherStaticRepositoryInstantiates()
        {
            var repo = new WeatherStaticRepository();
            Approvals.VerifyAll(repo, "Weather Point");
        }

        [TestMethod()]
        public void WeatherStaticRepositoryHasCorrectMinimumTemperatureDifferential()
        {
            var repo = new WeatherStaticRepository();
            var result =
                from w in repo
                where -w.TemperatureDifferential ==
                    repo.Min(d => -d.TemperatureDifferential)
                select w;
                         
            Approvals.VerifyAll(result, "Weather Data Point");
        }
    }
}