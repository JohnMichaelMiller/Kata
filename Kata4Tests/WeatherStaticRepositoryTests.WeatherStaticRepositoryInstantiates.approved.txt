Weather Point[0] = Day:1 Min Temp: 59, Max Temp 59, Temp Diff: -29
Weather Point[1] = Day:2 Min Temp: 59, Max Temp 59, Temp Diff: -20
Weather Point[2] = Day:3 Min Temp: 55, Max Temp 55, Temp Diff: -22
Weather Point[3] = Day:4 Min Temp: 59, Max Temp 59, Temp Diff: -18
Weather Point[4] = Day:5 Min Temp: 66, Max Temp 66, Temp Diff: -24
